pkg/controller/certificates/issuing/issuing_controller.go

```diff
@@ -238,14 +238,24 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 
 	cond := apiutil.GetCertificateRequestCondition(req, cmapi.CertificateRequestConditionReady)
 	if cond == nil {
+		// If the certificate request has been denied, set the last failure time to
+		// now, and set the Issuing status condition to False with reason. We only
+		// perform this check if the request also doesn't have a Ready condition,
+		// since some issuers may not honor a Denied condition, and will sign and
+		// set the Ready condition to True anyway. We would still want to complete
+		// issuance for requests where the issuer doesn't respect approval.
```

Comment:
I re-created another comment since my initial comment (https://github.com/jetstack/cert-manager/pull/3795/files#r600322397) has drifted to another discussion 😅

I suggest focusing on the "why" rather than "what" in the above comment, i.e., the above comment should start with the "why" since that's what people are looking for.

The "what" should only be written when the code itself is hard to understand, or too long, or too tricky. I don't think this block of code is tricky/long/hard to understand, so there shouldn't need to comment on the "what".

Here is a suggestion of comment that only explains the "why":

```suggestion
		// Some issuers won't honor the Denied=True condition, and we don't
		// want to break these issuers. To avoid breaking these issuers, we skip
		// bubbling up the Denied=True condition from the request to the
		// certificate when the issuer seems to have ignored the Denied
		// condition and has set the Ready condition to True or False anyways.
```